powerauth-fido2/src/test/java/com/wultra/powerauth/fido2/rest/model/converter/AssertionChallengeConverterTest.java

```diff
@@ -0,0 +1,168 @@
+        operationDetailResponse.setApplications(List.of("app"));
+        operationDetailResponse.setId("operationID");
+        operationDetailResponse.setData("A1*A100CZK");
+        operationDetailResponse.setFailureCount(0L);
+        operationDetailResponse.setMaxFailureCount(5L);
+
+        final AssertionChallenge assertionChallenge = AssertionChallengeConverter
+                .convertAssertionChallengeFromOperationDetail(operationDetailResponse, Collections.emptyList());
+        assertEquals("user", assertionChallenge.getUserId());
+        assertEquals("app", assertionChallenge.getApplicationIds().get(0));
+        assertEquals("operationID&A1*A100CZK", assertionChallenge.getChallenge());
+        assertEquals(0L, assertionChallenge.getFailedAttempts());
+        assertEquals(5L, assertionChallenge.getMaxFailedAttempts());
+        assertNull(assertionChallenge.getAllowCredentials());
+    }
+
+    @Test
+    void testConvertAssertionChallengeFromOperationDetail_nonWultraAuthenticatorDetail() {
+        final OperationDetailResponse operationDetailResponse = new OperationDetailResponse();
+        operationDetailResponse.setUserId("user");
+        operationDetailResponse.setApplications(List.of("app"));
+        operationDetailResponse.setId("operationID");
+        operationDetailResponse.setData("A1*A100CZK");
+        operationDetailResponse.setFailureCount(0L);
+        operationDetailResponse.setMaxFailureCount(5L);
+
+        final AuthenticatorDetail authenticatorDetail = new AuthenticatorDetail();
+        authenticatorDetail.setCredentialId(Base64.getEncoder().encodeToString("credential-1".getBytes()));
+        authenticatorDetail.setExtras(Map.of(
+                "transports", List.of("hybrid"),
+                "aaguid", "00000000-0000-0000-0000-000000000000"));
+        final List<AuthenticatorDetail> authenticatorDetails = List.of(authenticatorDetail);
+
+        final AssertionChallenge assertionChallenge = AssertionChallengeConverter
+                .convertAssertionChallengeFromOperationDetail(operationDetailResponse, authenticatorDetails);
+        assertEquals("user", assertionChallenge.getUserId());
+        assertEquals("app", assertionChallenge.getApplicationIds().get(0));
+        assertEquals("operationID&A1*A100CZK", assertionChallenge.getChallenge());
+        assertEquals(0L, assertionChallenge.getFailedAttempts());
+        assertEquals(5L, assertionChallenge.getMaxFailedAttempts());
+
+        assertNotNull(assertionChallenge.getAllowCredentials());
+        final AllowCredentials allowCredential = assertionChallenge.getAllowCredentials().get(0);
+        assertArrayEquals("credential-1".getBytes(), allowCredential.getCredentialId());
+        assertEquals("hybrid", allowCredential.getTransports().get(0));
+        assertEquals("public-key", allowCredential.getType());
+    }
+
+    @Test
+    void testConvertAssertionChallengeFromOperationDetail_withWultraAuthenticatorDetail() {
```

Comment:
Have you considered a test case for multiple Wultra authentication details?

Reply:
good idea, I added the test